spec/models/event_spec.rb

```diff
@@ -13,4 +13,10 @@
 
     it { expect(described_class.search("event")).to contain_exactly(match_beginning, match_middle, match_end) }
   end
+
+  describe "#validate_end_on_before_start_on" do
+    it "does not allow end date to be before start date" do
+      expect(FactoryGirl.build(:event, name: "End Date before Start Date", start_on: 1.year.ago + 1.day, end_on: 1.year.ago)).to_not be_valid
```

Comment:
I tried `let` initially, but as the input is invalid the `let` is not being created and it throws an error.